### ISSUE TITLE
`TypeError: can't access property "label",` when editing a record

### ISSUE THREAD

**Issue by grantfitzsimmons** [specify/specify7#2949]
This does not happen if the user is an administrator.

Specify 7 version: `v7.7.1`

Attempting to edit this record:
https://specify.bayworld.saiab.ac.za/specify/view/collectionobject/166178/

All necessary permissions are present:

<img width="555" alt="image" src="https://user-images.githubusercontent.com/37256050/217274833-6565dd3a-e758-4ee6-8cf9-00d58d148b3b.png">

Somehow it seems there is a permission issue when updating the records:

<img width="681" alt="image" src="https://user-images.githubusercontent.com/37256050/217275499-b09f2be5-3363-4d94-b1a9-d9faed25029a.png">

This user has full access to all tables.

They can create a new collection object, but not edit one they just created.

**Console Logs:**

```
"type": "permissionDenied",
  "statusText": "You don't have a permission to do this action",
  "responseText": "{\"NoMatchingRuleException\": [{\"collectionid\": 131072, \"userid\": 10, \"resource\": \"/field/collectionobject/fieldnumber\", \"action\": \"update\"}]}"
```

```
TypeError: can't access property "label", u(...)[(n.length - 1)][n.slice(...)[0]] is undefined
```

[Specify 7 Crash Report - 2023-02-07T14 36 08.191Z.txt](https://github.com/specify/specify7/files/10676349/Specify.7.Crash.Report.-.2023-02-07T14.36.08.191Z.txt)

**Reported By:** Willem at SAIAB

**Comment by maxxxxxdlp** [comment]
TLDR: duplciate of https://github.com/specify/specify7/issues/2394

---

Long response:

Looking at the crash report, can confirm that user has update permission to the determination table in the collection with id 131072.

However, the errors speaks about the permission to edit the "text1" field in the determination table. This is because back-end already has support for field-level permissions. Front-end for now just.

This issue is a duplicate of https://github.com/specify/specify7/issues/2394

**Comment by grantfitzsimmons** [comment]
@maxxxxxdlp Can you propose a solution for SAIAB at the moment?

**Comment by maxxxxxdlp** [comment]
A proposed solution is in https://github.com/specify/specify7/issues/2394#issuecomment-1319010501
See also https://github.com/specify/specify7/issues/2394#issuecomment-1421294395
